datafusion/common/src/hash_utils.rs

```diff
@@ -692,6 +730,48 @@ mod tests {
         assert_eq!(hashes[0], hashes[1]);
     }
 
+    #[test]
+    // Tests actual values of hashes, which are different if forcing collisions
+    #[cfg(not(feature = "force_hash_collisions"))]
+    fn create_hashes_for_map_arrays() {
```

Comment:
It would help me undertand / verify this test if you could use a `MapBuilder` or add a comment showing what MapArray was being built

Reply:
done! https://github.com/apache/datafusion/pull/11547/commits/06b7d3cc8a96b7d1b38277de14326303b3b47401